api/v1beta1/glance_types.go

```diff
@@ -99,12 +99,10 @@ type GlanceSpec struct {
 	StorageRequest string `json:"storageRequest"`
 
 	// +kubebuilder:validation:Required
-	// +kubebuilder:default:={replicas: 1}
 	// GlanceAPIInternal - Spec definition for the internal and admin API service of this Glance deployment
```

Comment:
This was basically a contradiction. You cannot say GlanceAPIInternal is a required field and also at the same time provide some default values for it. As far as I understand this combination of kubebuilder markers made the GlanceAPIInternal field still optional with the default (i.e. Required was ignored).

At the same time GlanceAPISpec.ContainerImage is marked as Required[1]. Which means creating a GlanceAPISpec with the value `{replicas: 1}` is not possible as `containerImage` is not specified in that value but the field is Required. This caused the actual failure.

[1] https://github.com/openstack-k8s-operators/glance-operator/blob/e8a8dd567b252af6e89b6d9bd000c36db846c76c/api/v1beta1/glanceapi_types.go#L41-L43